react-native/react/menubar/index.render.desktop.js

```diff
@@ -200,49 +228,84 @@ class FolderList extends Component {
 
     // Remove folders that are just our personal ones, we'll add those in later
     // For consistency. Since we aren't gauranteed we have favorited our own folders.
-    const folders = this.props.folders.filter(f => stripPublicTag(f.folderName) !== username)
-
-    const personalPrivateFolder: FolderInfo = {
-      folderName: username,
-      isPublic: false,
-      isEmpty: true,
-      openFolder: this.props.openKBFSPrivate
+    const folders = this.props.folders.filter(f => f.type === 'entry' || stripPublicTag(f.folderName) !== username)
+
+    let privateFolders = []
+    let publicFolders = []
+
+    const isLoggedIn = true // TODO plumb this through
+
+    if (isLoggedIn) {
+      const personalPrivateFolder: FolderInfo = {
+        type: 'folder',
+        folderName: username,
+        isPublic: false,
+        isEmpty: true,
+        openFolder: () => this.props.openKBFSPrivate(username)
+      }
+      privateFolders.push(personalPrivateFolder)
+
+      const privateFolderEntry: FolderInfo = {
+        type: 'entry',
+        isPublic: false,
+        prefix: `${username},`,
+        openFolder: folder => this.props.openKBFSPrivate(`${username},${folder}`)
+      }
+      privateFolders.push(privateFolderEntry)
     }
 
     const personalPublicFolder: FolderInfo = {
+      type: 'folder',
       folderName: username,
       isPublic: true,
       isEmpty: true,
-      openFolder: this.props.openKBFSPublic
+      openFolder: () => this.props.openKBFSPublic(username)
+    }
+
+    const publicFolderEntry: FolderInfo = {
+      type: 'entry',
+      isPublic: true,
+      prefix: '',
+      openFolder: folder => this.props.openKBFSPublic(folder)
     }
 
-    const privateFolders = [personalPrivateFolder].concat(folders.filter(f => !f.isPublic))
-    const publicFolders = [personalPublicFolder].concat(folders.filter(f => f.isPublic)).map(f => ({...f, folderName: stripPublicTag(f.folderName)}))
+    publicFolders.push(personalPublicFolder)
+    publicFolders.push(publicFolderEntry)
+
+    privateFolders = privateFolders.concat(folders.filter(f => !f.isPublic))
+    publicFolders = publicFolders.concat(
+    folders.filter(f => f.isPublic))
+    .map(f => f.type === 'entry' ? f : {...f, folderName: stripPublicTag(f.folderName)})
```

Comment:
I don't think you'll have an entry type here. You filtered that out when you made folders on L231. If you want a stronger guarantee you can declare folders to be type `Array<FolderFolder>` (assuming `FolderFolder` is `Folder` `type='folder'`).